pkg/workspace/resource/node.go

```diff
@@ -127,23 +140,84 @@ func (c *NodeManager) getReadyNodesFromNodeClaims(ctx context.Context, wObj *kai
+			return false, fmt.Errorf("failed to update Node status condition(NodesReady): %w", updateErr)
+		}
+		return true, nil
+	} else {
+		klog.InfoS("Ready Nodes for workspace are not enough currently", "workspace", client.ObjectKeyFromObject(wObj).String(),
+			"targetNodes", targetNodeCount, "currentReadyNodes", readyCount)
+		if updateErr := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, kaitov1beta1.ConditionTypeNodeStatus, metav1.ConditionFalse,
+			"NodeNotReady", fmt.Sprintf("Ready Nodes are not enough (TargetNodes: %d, CurrentReadyNodes: %d, SelectedNodes: %d)", targetNodeCount, readyCount, len(nodeList.Items))); updateErr != nil {
+			klog.ErrorS(updateErr, "failed to update Node status condition NodesReady to false", "workspace", klog.KObj(wObj))
+			return false, fmt.Errorf("failed to update Node status condition(NodeNotReady): %w", updateErr)
+		}
+
+		return false, nil
+	}
+}
+
+// SetConditionWithOwnedConditions checks the owned conditions, and if any one is false, it sets the condition to false with the reason and message of the first false condition found.
+func (c *NodeManager) SetConditionWithOwnedConditions(ctx context.Context, wObj *kaitov1beta1.Workspace, condition kaitov1beta1.ConditionType, conditionTypes []kaitov1beta1.ConditionType) error {
+	for _, cType := range conditionTypes {
+		ownedCondition := meta.FindStatusCondition(wObj.Status.Conditions, string(cType))
+		if ownedCondition == nil {
+			continue // Condition not found, skip to the next one
+		} else if ownedCondition.Status != metav1.ConditionTrue {
+			// Set the owned condition to false with the reason and message of the first false condition found.
+			if err := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, condition, metav1.ConditionFalse, ownedCondition.Reason, ownedCondition.Message); err != nil {
+				klog.ErrorS(err, "failed to update workspace status", "workspace", klog.KObj(wObj))
+				return err
 			}
-			return err
+			return nil
 		}
 	}
 
+	return nil
+}
+
+// SetResourceReadyConditionByStatus updates the status of the resource ready condition based on the statuses of owned conditions.
+func (c *NodeManager) SetResourceReadyConditionByStatus(ctx context.Context, wObj *kaitov1beta1.Workspace) error {
+	ownedConditions := []kaitov1beta1.ConditionType{
+		kaitov1beta1.ConditionTypeNodeClaimStatus,
+		kaitov1beta1.ConditionTypeNodeStatus,
+		kaitov1beta1.ConditionTypeNodePluginStatus,
+	}
+
+	// If any owned condition is false, set the resource condition to false and return.
+	if err := c.SetConditionWithOwnedConditions(ctx, wObj, kaitov1beta1.ConditionTypeResourceStatus, ownedConditions); err != nil {
+		return err
+	}
+
+	// All owned conditions are true, set the resource condition to true.
 	if err := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, kaitov1beta1.ConditionTypeResourceStatus, metav1.ConditionTrue,
```

Comment:
It seems that resource condition will be set to true even if all owned conditions are false.